__tests__/endpoints.test.js

```diff
@@ -137,30 +137,90 @@ describe("GET /api/articles", () => {
+      return request(app)
+        .get("/api/articles?topic=notopic")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not found");
+        });
+    });
 
-  test("400: should respond with a 400 status code if either or both of the query parameters do not do not coincide with what is allowed in the valid columns", () => {
-    return request(app)
-      .get("/api/articles?sort_by=author&order_by=incorrectOrder")
-      .expect(400)
-      .then(({ body }) => {
-        expect(body.msg).toBe("Invalid order query");
-      });
-  });
-  test("400: should respond with a 400 status code if either or both of the query parameters do not coincide with what is allowed in the valid columns", () => {
-    return request(app)
-      .get("/api/articles?sort_by=incorrect&order_by=asc")
-      .expect(400)
-      .then(({ body }) => {
-        expect(body.msg).toBe("Invalid sort_by column");
-      });
+    test("400: should respond with a 400 status code if either or both of the query parameters do not do not coincide with what is allowed in the valid columns", () => {
+      return request(app)
+        .get("/api/articles?sort_by=author&order_by=incorrectOrder")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid order query");
+        });
+    });
+    test("400: should respond with a 400 status code if either or both of the query parameters do not coincide with what is allowed in the valid columns", () => {
+      return request(app)
+        .get("/api/articles?sort_by=incorrect&order_by=asc")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid sort_by column");
+        });
+    });
+    //topics queries
+    test("200: should return articles by topic in acesdening order", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body.articles;
+          const sortExpected = _.orderBy(articles, ["topic"], ["asc"]);
+          expect(articles).toEqual(sortExpected);
+        });
+    });
+    test("200: should return articles by topic in decsending order", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch&order_by=desc")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body.articles;
+          const sortExpected = _.orderBy(articles, ["topic"], ["desc"]);
+          expect(articles).toEqual(sortExpected);
+        });
+    });
+    //////
+    test("both querys can be used in tandom", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch&author=icellusedkars")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body.articles;
+          expect(articles[0].author).toEqual("icellusedkars");
+          expect(articles[0].topic).toEqual("mitch");
```

Comment:
could still iterate over all.  there's an outside chance this test could pass with the wrong input being sent back.